omod/src/main/java/org/openmrs/module/queue/web/resources/QueueEntryCountSubResource.java

```diff
@@ -52,7 +52,7 @@ public void setParent(QueueEntryCount queueEntryCount, Queue queue) {
 	@Override
 	public PageableResult doGetAll(Queue queue, RequestContext requestContext) throws ResponseException {
 		return new GenericSingleObjectResult(Arrays.asList(new PropValue("queueName", queue.getName()),
-		    new PropValue("queueEntriesCount", queue.getQueueEntries().size())));
+		    new PropValue("queueEntriesCount", queue.getActiveQueueEntries().size())));
```

Comment:
This change is needed for this to replicate the existing functionality, but is this what we want?  Should the "getAll" resource limit itself to only active queue entries?  Or should getAll really ...get all.  And then we can limit this to only active entries by passing in a filter or search parameter.

Reply:
This is a weird resource. Just returns the counts. A response like:
```java
{
  "QueueName": "Triage Queue",
  "count": "36"
}
```
The idea is to get the number of patients in a queue. Mainly used in the UI to display totals for queues.

Reply:
Hmmm... It doesn't seem like this resource is going to perform in an ideal way. I'd prefer it if we had one endpoint that returned "the current queue", i.e., queue name, all active entries, and counts. Then we could use the custom view code if someone really want to limit things down to just requesting, e.g., the name and count. This just encourages a pattern of access where we're making a lot of REST requests, which is bad for performance.

@mseaton In the context of a queue, I think it's fine if the default is to return active entries only. It seems reasonable to have a query property that be used to also request the queue with inactive entries, but I'm not clear on the use-case for that. This may be a little different from other domains, but I tend to think of inactive queue entries as similar to retired metadata.

Reply:
I understand the primary use case.  A potentially more comprehensive endpoint might return an object that contains data that is a bit more complete and includes counts as the service/api deals with them, so it's clear.  For example, if our "getActiveQueueEntries" filters out voided, started-in-future, and ended entries (which it does), then we could ensure that all of these are represented in this endpoint comprehensively like:
```
{
  "QueueName": "Triage Queue",
  "totalCount": "63",
  "activeCount": "36",
  "futureCount": "5",
  "voidedCount": "2",
  "endedCount": "20"
}
```

Reply:
Sorry, I made this comment before I saw @ibacher  response.  Must have been typing at the same time.  You can disregard my comment, especially if we get rid of this endpoint as Ian seems to be suggesting :)

Reply:
> This may be a little different from other domains, but I tend to think of inactive queue entries as similar to retired metadata

I understand your point and I'm fine if the endpoint returns active entries by default.  I just wanted to bring it up to make sure.  As you indicate, we can always add support for query parameters like "includeEnded=true", "includeVoided=true", and "includeFuture=true" for those use cases if/when they arise.

Reply:
Yeah, haven't dug into it in much detail but the use cases for get all "ended" queue members or a count of all of them seems relatively small.  If I'm understand it correctly the "ended" queue entries will just continue to grow and grow indefinitely and the general need when asking for "all" would be "get me all currently in the queue".  There might be a case to say, "get me all cases when patient X was in this queue in the past month" but "getAllEntires" seems about as useful as "getAllObs".  (Again, If I'm understanding this all correctly).